### PR TITLE
Add NimBLE heart rate sensor example (GATT notifications)

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
This PR adds an example application for configuring a RIOT node as BLE heart rate sensor using NimBLE. The main focus of this application is to demonstrate the usage of NimBLE for asynchronous data transfers using GATT notifications (as the NimBLE API for this is not too intuitive...).

### Testing procedure
Flash this example to your favorite nrf-based board (e.g. `nrf52dk`) and connect to the node using any heart rate sensor capable Android/iOS app. For Android you could use e.g. Nordics 'nRF Toolbox` (https://play.google.com/store/apps/details?id=no.nordicsemi.android.nrftoolbox)

### Issues/PRs references
none

### CODE REVIEW

**Comment by aabadie** [comment]
I briefly tested this PR on nrf52840-mdk following the testing procedure and it was working.
At the beginning nRF toolbox wasn't able to detect the RIOT device but after some time I could connect and received HR notifications.

This PR can be a good base for testing notfications of other GATT services/characteristics.

**Comment by haukepetersen** [comment]
So would you be good with merging this?

File: examples/nimble_heart_rate_sensor/Makefile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# We also use the AD part of the BLE helper module
+USEMODULE += bluetil_ad
+
+# Comment this out to disable code in RIOT that does safety checking
+# which is not needed in a production environment but helps in the
+# development process:
+DEVELHELP ?= 1
+
+# REMOVE
+CFLAGS += -DDEBUG_ASSERT_VERBOSE
```
Comment (aabadie) [inline examples/nimble_heart_rate_sensor/Makefile]: I guess this should be removed before merging

File: sys/include/net/ble.h [hunk 2]
```diff
@@ -85,6 +85,39 @@ extern "C" {
...
+ * @name    Selected GATT characteristic UUIDs (16-bit)
+ *
+ * @see https://www.bluetooth.com/specifications/gatt/characteristics/
+ * @{
+ */
+#define BLE_GATT_CHAR_BATTERY_LEVEL         (0x2a19)    /**< battery level */
+#define BLE_GATT_CHAR_SYSTEM_ID             (0x2a23)    /**< system ID */
+#define BLE_GATT_CHAR_MODEL_NUMBER_STR      (0x2a24)    /**< model number */
+#define BLE_GATT_CHAR_SERIAL_NUMBER_STR     (0x2a25)    /**< serial number */
+#define BLE_GATT_CHAR_FW_REV_STR            (0x2A26)    /**< firmware revision */
```
Comment (aabadie) [inline sys/include/net/ble.h]: For consistency, maybe use lower case `A` in hex value ?

File: examples/nimble_heart_rate_sensor/Makefile (hunk 1 above)
Comment (haukepetersen) [reply examples/nimble_heart_rate_sensor/Makefile]: yes, missed that one. Good thing I did put the comment to it :-)

File: sys/include/net/ble.h (hunk 2 above)
Comment (haukepetersen) [reply sys/include/net/ble.h]: yes, copy-paste typo, will fix.

**Comment by haukepetersen** [comment]
addressed comments, and squashed.

**Comment by haukepetersen** [comment]
thanks for the review :-)

**Pull request merged**